var/spack/repos/builtin/packages/arrayfire/package.py

```diff
@@ -64,10 +61,34 @@ def cmake_args(self):
             self.define_from_variant('AF_BUILD_CUDA', 'cuda'),
             self.define_from_variant('AF_BUILD_FORGE', 'forge'),
             self.define_from_variant('AF_BUILD_OPENCL', 'opencl'),
+            self.define_from_variant('BUILD_TESTING', 'tests'),
         ])
+
+        if '+cuda' in self.spec:
+            cuda_arch_list = self.spec.variants['cuda_arch'].value
+            cuda_arch = cuda_arch_list[0]
+            if cuda_arch == 'none':
+                args.append(self.define('CUDA_architecture_build_targets', 'Auto'))
+            else:
+                arch_list_string = []
+                for arch in cuda_arch_list:
+                    arch_list_string.append('{}.{}'.format(arch[:-1], arch[-1]))
```

Comment:
This is a good place for a list comprehension:
```suggestion
                arch_list_string = ['{}.{}'.format(arch[:-1], arch[-1]) for arch in cuda_arch_list]
```